apps/overige-object-api/src/utils/fetchData.ts

```diff
@@ -0,0 +1,146 @@
+import { ErrorHandler } from './errorHandler';
+
+interface HandleGraphqlRequestProps {
+  query?: string;
+  variables: any;
+  headers?: HeadersInit;
+}
+
+const handleGraphqlRequest = ({ query, variables, headers }: HandleGraphqlRequestProps) =>
+  ({
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      ...headers, // Merge custom headers (including the ability to overwrite 'Content-Type')
+    },
+    body: JSON.stringify({ query, variables }),
+    cache: 'no-store',
+  }) as RequestInit;
+
+export interface FetchDataProps {
+  url: string;
+  query?: string;
+  variables?: any;
+  method?: string;
+  headers?: HeadersInit; // Allow custom headers to be passed
+}
+
+/**
+ * @description Fetches data from the server (GraphQL or REST).
+ * @param {string} url - The URL to fetch data from.
+ * @param {string} query - The GraphQL query (if applicable).
+ * @param {any} variables - The variables to pass to the GraphQL queries.
+ * @param {string} method - The HTTP method, default is POST for GraphQL.
+ * @param {HeadersInit} headers - Custom headers to pass to the request.
+ * @returns {Promise<T>} - The fetched data.
+ */
+export const fetchData = async <T>({
+  url,
+  query,
+  variables,
+  method = 'POST',
+  headers = {}, // Default to an empty object if no headers are provided
+}: FetchDataProps): Promise<T> => {
+  // Default headers, which can be overwritten by custom headers (e.g., Content-Type)
+  const defaultHeaders: HeadersInit = {
+    'Content-Type': 'application/json',
+  };
+
+  const requestOptions: RequestInit = query
+    ? handleGraphqlRequest({ query, variables, headers: { ...defaultHeaders, ...headers } })
+    : {
+        method,
+        cache: 'no-store',
+        headers: {
+          ...defaultHeaders,
+          ...headers, // Merge custom headers with default ones (overwriting defaults if needed)
+        },
+      };
+
+  try {
+    const response = await fetch(url, requestOptions);
+
+    // Check for non-successful responses (status not in the 2xx range)
+    if (!response.ok) {
+      handleHttpError(response);
+    }
+
+    const data = await response.json();
+
+    // Handle GraphQL-specific errors
+    if (data.errors && data.errors.length > 0) {
+      data.errors.forEach(handleGraphqlError); // Process each error
+    }
```

Comment:
De volgende comments zijn echt heel erg stating the obvious, wat mij betreft kunnen ze weg.

- `// Default to an empty object if no headers are provided`
- `// Check for non-successful responses (status not in the 2xx range)`
- `// Process each error`